src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -0,0 +1,1138 @@
+                TestPlatformOptions = options,
+                TestSessionInfo = testSessionInfo
+            };
+
+            // TODO: this will need to be via an adapter instead. Because we will get a "live"
+            // implementation rather than our own implementation that we normally have in design
+            // mode client. It probably will even throw on this cast.
+            var upcastCustomLauncher = (ITestHostLauncher3?)customLauncher;
+
+            TestRequestManager?.RunTests(
+                testRunPayload,
+                upcastCustomLauncher,
+                new RunHandlerToEventsRegistrarAdapter(testRunEventsHandler),
+                new ProtocolConfig { Version = _highestSupportedVersion });
+        }
+        catch (Exception ex)
+        {
+            EqtTrace.Error("DesignModeClient: Exception in StartTestRun: " + ex);
+            var testRunCompleteArgs = new TestRunCompleteEventArgs(null, false, true, ex, null, null, TimeSpan.MinValue);
+
+            testRunEventsHandler.HandleLogMessage(TestMessageLevel.Error, ex.ToString());
+            testRunEventsHandler.HandleTestRunComplete(testRunCompleteArgs, null, null, null);
+        }
+
+        _testPlatformEventSource.TranslationLayerExecutionStop();
+    }
+
+    #region Async, not implemented
+    /// <inheritdoc/>
+    public async Task DiscoverTestsAsync(
+        IEnumerable<string> sources,
+        string? discoverySettings,
+        ITestDiscoveryEventsHandler discoveryEventsHandler)
+    {
+        await DiscoverTestsAsync(
+                sources,
+                discoverySettings,
+                options: null,
+                new DiscoveryEventsHandleConverter(discoveryEventsHandler))
+            .ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async Task DiscoverTestsAsync(
+        IEnumerable<string> sources,
+        string? discoverySettings,
+        TestPlatformOptions? options,
+        ITestDiscoveryEventsHandler2 discoveryEventsHandler)
+    {
+        await DiscoverTestsAsync(
+                sources,
+                discoverySettings,
+                options,
+                testSessionInfo: null,
+                discoveryEventsHandler)
+            .ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async Task DiscoverTestsAsync(
+        IEnumerable<string> sources,
+        string? discoverySettings,
+        TestPlatformOptions? options,
+        TestSessionInfo? testSessionInfo,
+        ITestDiscoveryEventsHandler2 discoveryEventsHandler)
+    {
+        // The Task.Delay(100) is a dumb way of silencing the analyzers, otherwise an error saying
+        // the method lacks an await statement will pop up. This is only temporary though, until a
+        // working implementation of this method will completely replace the current method body.
+        await Task.Delay(100, CancellationToken.None);
+        throw new NotImplementedException();
+    }
```

Comment:
Yep, this was the way it was before adding the `async` keyword, and even though it's not a change to the API, I didn't like it because it had some (minimal) error potential. Changing according to suggestion though.